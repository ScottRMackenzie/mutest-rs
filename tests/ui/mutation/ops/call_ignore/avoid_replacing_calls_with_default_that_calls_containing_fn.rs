//@ print-mutations
//@ build
//@ stdout
//@ stderr: empty
//@ mutation-operators: call_delete, call_value_default_shadow

#![allow(unused)]

struct S {
    v: usize,
    b: bool,
}

impl Default for S {
    fn default() -> Self {
        S::new_with_args(0, false)
    }
}

impl S {
    fn new(_: ()) -> Self {
        Self { v: 0, b: false }
    }

    fn new_with_args(v: usize, b: bool) -> Self {
        let mut s = S::new(());
        s.v = v;
        s.b = b;
        s
    }
}

fn f() {
    let default_s = S::new_with_args(0, false);
}

#[test]
fn test() {
    f();
}
