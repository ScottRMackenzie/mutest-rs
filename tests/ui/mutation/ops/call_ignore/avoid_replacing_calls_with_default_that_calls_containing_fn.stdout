1 mutation
  - [call_delete] delete call to `S::new_with_args` and replace it with `Default::default()` in f at tests/ui/mutation/ops/call_ignore/avoid_replacing_calls_with_default_that_calls_containing_fn.rs:34:21: 34:47
    <-(0)- test

1 mutation
  - [call_value_default_shadow] ignore return value of call to `S::new_with_args` by shadowing it with `Default::default()` in f at tests/ui/mutation/ops/call_ignore/avoid_replacing_calls_with_default_that_calls_containing_fn.rs:34:21: 34:47
    <-(0)- test

2 mutants; 2 mutations; 2 safe; 0 unsafe (0 tainted); 0 batched; 2 unbatched
